Adjust source map paths for .mjs

jaketrent opened the pull request:
Resolves: #5

Thanks for the tips in the issue on how you'd like this approached. They were helpful.

I'd love for this to get merged.

I tried two different methods. The first was using magicString.generateMap to just regenerate source maps, but I switched to this approach because it seemed simpler and less invasive.

jaketrent committed:
Adjust source map paths for .mjs

Resolves: #5

jaketrent commented on test/outdir-convert/valid.test.mjs:
```diff
@@ -3,44 +3,120 @@ import * as assert from 'uvu/assert';
 import { extname, basename, resolve } from 'path';
```
There are more changes to the test suite than I wanted to make initially for this feature add. Considering there are now two kinds of things being formatted and tested (.mjs and .mjs.map), I decided to make organizational changes to more clearly express what was being tested.

kristoferbaxter replied:
First take here is spitting the concerns into different files for the api versus cli might make sense.

I'm going to grab this branch locally and take a look at what this looks like.

jaketrent committed:
prettier format

jaketrent commented:
@kristoferbaxter any feedback on this?

kristoferbaxter commented:
Apologies, I missed this. Reviewing now.

kristoferbaxter commented on src/format.mjs:
```diff
@@ -61,17 +61,25 @@ export async function format(configFileLocation, config) {
+        newFilePaths.add(newFilePath + '.map');
+        await fs.writeFile(newFilePath + '.map', sourceMap);
+        await fs.unlink(filePath + '.map');
+      }
```
This all looks good and clear to me.

kristoferbaxter commented on src/format.mjs:
```diff
@@ -80,3 +88,18 @@ export async function format(configFileLocation, config) {
   return newFilePaths;
 }
+
+function convertSourceMapURL(filePath, fileContents) {
```
The larger this project is, the more I am considering using TypeScript for authoring it. This might not be the right place, but soon.

kristoferbaxter commented on test/prettier.mjs:
```diff
@@ -1,15 +1,13 @@
  * @return {string}
  */
-export async function formatter(content) {
+export async function formatter(content, prettierOptions) {
```
This change moves away from a singleton, intended?

jaketrent replied:
Yes, it was intended in order to allow for parameterized options in test, where the json parser is used in some cases. But if there is some other side effect of the change, that may not have been intended. I see that I forgot to remove line 5 at the very least, which is now a stale comment.

kristoferbaxter commented on yarn.lock:
```diff
@@ -677,6 +687,31 @@ locate-path@^6.0.0:
   dependencies:
     p-locate "^5.0.0"
 
+lodash._reinterpolate@^3.0.0:
```
Hmm, looks like `sourcemap-validator` pulls in quite a few items. Not an issue, but something we might be able to upstream changes for.

kristoferbaxter reviewed (approved):
Approving to unblock, will grab locally to see if splitting the tests across two files makes more sense.

kristoferbaxter commented on test/outdir-convert/valid.test.mjs:
```diff
@@ -3,44 +3,120 @@ import * as assert from 'uvu/assert';
+
+cli.before(async (context) => {
+  await compileTypescript();
+  context.formattedFiles = await formatViaApi();
```
Believe this was supposed to be `formatViaCli()`.

jaketrent committed:
remove stale comment

jaketrent committed:
split valid tests into api and cli files; find assertion error

jaketrent committed:
fix tsc target error

test/outdir-convert/tsconfig.json:9:15 - error TS6046:
Argument for '--target' option must be: 'es3', 'es5', 'es6', 'es2015', 'es2016', 'es2017', 'es2018', 'esnext'.

jaketrent committed:
update fixtures after assertion error discovery

jaketrent committed:
prettier format

jaketrent commented on test/outdir-convert/valid.mjs:
```diff
@@ -0,0 +1,38 @@
+    const fixtureFileSystem = await fs.readFile(resolve('test/outdir-convert/fixtures', basename(formattedEntry)), 'utf8');
+    const input = await formatter(formattedFileSystem, prettierOptions)
+    const fixture = await formatter(fixtureFileSystem, prettierOptions)
+    assert.equal(input, fixture);
```
Discovered a bug in test, both in this branch and master, that was comparing promises instead of awaited values. This should fix. Also uncovered that the fixtures needed updating too. That came in a later commit.

jaketrent commented:
Responded in comments, made the test adjustment, discovered a bug, and ready for a re-review.

kristoferbaxter commented:
Awesome, I made a few changes on a local copy of this branch. Would you mind if I merged them in too?

kristoferbaxter commented:
I created #13 that uses this as the base with a slightly different structure for the tests to maintain the tests as consistent independent of the before execution steps.
